cmd/podman/kube/apply.go

```diff
@@ -0,0 +1,112 @@
+package kube
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/cmd/podman/utils"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/spf13/cobra"
+)
+
+var (
+	applyOptions     = entities.ApplyOptions{}
+	applyDescription = `Command applies a podman container, pod, volume, or kube yaml to a Kubernetes cluster when a kubeconfig file is given.`
+
+	applyCmd = &cobra.Command{
+		Use:               "apply [options] {CONTAINER...|POD...|VOLUME...}",
+		Short:             "Deploy a podman container, pod, volume, or Kubernetes yaml to a Kubernetes cluster",
+		Long:              applyDescription,
+		RunE:              apply,
+		Args:              cobra.MinimumNArgs(0),
```

Comment:
cli syntax does not match validation,

If I read it correctly you want to accept 0 to n args, so at that point we could just skip all validation and remove the `Args` function because `MinimumNArgs(0)` is a NOP.

Also `{}` means one arg is mandatory but this is wrong it should be `[]` because it is optional.
The IMO best way would be `[CONTAINER...|POD...|VOLUME...]`

Reply:
fixed